### PR TITLE
Write blog post on fast goto definition.

### PR DESCRIPTION
**Pull request by olafurpg**
Live demo: https://olafurpg.github.io/metals/blog/2018/12/12/fast-goto-definition

### CODE REVIEW

File: NOTICE.md [hunk 1]
```diff
@@ -198,6 +198,11 @@ We include the text of the original license below:
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
-   
+
 ```
 
+# License notice for closeau
```
Comment (sjrd) [inline NOTICE.md]: typo: clouseau

File: website/blog/2018-12-12-index.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+`scala.Some.isEmpty` and many thousand source files with millions of lines of
+code, how do we quickly find the source file that defines that method?
+
+![goto-definition](https://user-images.githubusercontent.com/1408093/49591684-67070700-f96f-11e8-873d-90c40480528b.gif)
+
+There are some hard constraints:
+
+- we must answer quickly, normal requests should respond within 100-200ms.
+- memory usage should not exceed 10-100Mb since we also need memory to implement
+  other features we're sharing the computer with other applications.
```
Comment (sjrd) [inline website/blog/2018-12-12-index.md]: This sentence is weird. Is it missing an "and" somewhere?

File: website/blog/2018-12-12-index.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+needs to know the toplevel classes and nothing more.
+
+The algorithm for this custom parser that only extracts toplevel classes goes
+roughly like this.
+
+- tokenize source file
+- on `package` keyword, record package name
+- on `class` and `trait` and `object` keywords, record toplevel class
+- on `(` and `[` and `{` delimiters, skip tokens until we find matching closing
+  delimiter
```
Comment (sjrd) [inline website/blog/2018-12-12-index.md]: What about braced package declarations:
```scala
package foo

package bar {
  class Foobar
}
```
Also: classes in `package object`s?
?

Comment (olafurpg) [reply website/blog/2018-12-12-index.md]: Those are handled, the logic in this post is maybe a bit oversimplified. I'll update to make it clearer

File: website/blog/2018-12-12-fast-goto-definition.md [hunk 4]
```diff
@@ -116,21 +125,25 @@ val index = Map[Symbol, Path](
 )
 ```
 
-Our challenge is to build this index as fast as possible.
+With this index, we can lookup which file defines `scala.Some` and then parse
+`Option.scala` on-demand (like we do for Java) to find the position where
+`def isEmpty: Boolean` is defined. The challenge is to efficiently build the
+index.
```
Comment (gabro) [inline website/blog/2018-12-12-fast-goto-definition.md]: I like the bullet point style as in the Java section:


> With index we can:
>
> - lookup which file defines `scala.Some`
> - parse `Option.scala` on-demand (like we do for Java) to find the position where
`def isEmpty: Boolean` is defined
> 
> The challenge is to efficiently build the
index.

Comment (olafurpg) [reply website/blog/2018-12-12-fast-goto-definition.md]: Good idea, I will add bullets

**Pull request merged**